src/app/admin/services/controller/equipment-controller.service.ts

```diff
@@ -0,0 +1,225 @@
+      ) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  private isDateInPeriod(date: Date, period: UnavailableDates) {
+    const startUnavailable = new Date(period.start_date);
+    const endUnavailable = new Date(period.end_date);
+    const dateTime = date.getTime();
+    return date >= startUnavailable && date <= endUnavailable;
+  }
+
+  private archivateEquipment(data: TableAction<Equipment>) {
+    const equipment = data.row;
+    this.openArchiveEquipmentModal(equipment)
+      .pipe(
+        filter(Boolean),
+        switchMap(() => this.api.getEquipmentUnavailabilityPeriods(equipment.id)),
+        switchMap((res) => {
+          return res.items && res.items.length ? this.openOrderNotificationModal(data.action) : of(true);
+        }),
+        filter(Boolean),
+        switchMap(() => this.api.archiveEquipment(equipment.id)),
+        switchMap(() => this.fetchEquipments()),
+        untilDestroyed(this),
+      )
+      .subscribe(() => {
+        this.notificationService.openSuccess(`${equipment.title} было за архивированно!`);
+      });
+  }
+
+  private openArchiveConfirmation(name: string) {
+    // //todo add new logic
+    // return this.dialog
+    //   .open(ConfirmationModalComponent, {
+    //     ...this.commonModalConfig,
+    //     data: {
+    //       name,
+    //       title: ModalEnum.ArchiveTitle,
+    //       reason: ModalEnum.ArchiveReason,
+    //       applyButtonText: ModalEnum.ArchiveApplyButtonText,
+    //     },
+    //   })
+    //   .afterClosed()
+    //   .pipe(tap());
+  }
+
+  private openBlockConfirmation(name: string) {
+    // //todo add new logic
+    // return this.dialog
+    //   .open(ConfirmationModalComponent, {
+    //     ...this.commonModalConfig,
+    //     data: {
+    //       name,
+    //       title: ModalEnum.BlockTitle,
+    //       reason: ModalEnum.BlockReason,
+    //       applyButtonText: ModalEnum.BlockApplyButtonText,
+    //     },
+    //   })
+    //   .afterClosed();
+  }
+
+  private openOrderNotificationModal(action: string): Observable<unknown> {
+    return this.dialog
+      .open(OrderNotificationModalComponent, {
+        ...this.commonModalConfig,
+        data: action,
+      })
+      .afterClosed();
+  }
+
+  private openArchiveEquipmentModal(equipment: Equipment) {
+    return this.dialog
+      .open(ArchiveEquipmentModalComponent, {
+        ...this.commonModalConfig,
+        data: equipment,
+      })
+      .afterClosed();
+  }
+  private openBlockEquipmentModal(equipment: Equipment, unavailablePeriods: UnavailableDates[]): Observable<Period> {
+    return this.dialog
+      .open(BlockEquipmentModalComponent, {
+        ...this.commonModalConfig,
+        data: {
+          equipment,
+          unavailablePeriods: unavailablePeriods ? unavailablePeriods : [],
+        },
+      })
+      .afterClosed();
+  }
+
+  createDictionary(items: { id: number; name: string; translation?: string }[], dictionary: Dictionary<string>) {
```

Comment:
this may be a common helper. Also item type may have a separate file.

Reply:
done